### PR TITLE
#29 allow for an optional Clear option

### PR DESCRIPTION
**Pull request by reszli**

### CODE REVIEW

File: src/_shared/ModalDialog.jsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ const ModalDialog = (props) => {
       </DialogContent>
 
       <DialogActions>
+        {allowClear && <Button color="primary" onClick={onClear} tabIndex={-1}> Clear </Button>}
```
[the remark this answers is not in the document]
Comment (reszli) [reply src/_shared/ModalDialog.jsx]: You suggest removing the tabindex attribute completely or setting it to 0?

File: src/_shared/ModalDialog.jsx [hunk 2]
```diff
@@ -40,12 +41,16 @@ ModalDialog.propTypes = {
...
+  onClear: PropTypes.func,
+  allowClear: PropTypes.bool,
   classes: PropTypes.object.isRequired,
   dialogContentClassName: PropTypes.string,
 };
 
 ModalDialog.defaultProps = {
   dialogContentClassName: '',
+  allowClear: false,
+  onClear: null,
```
[the remark this answers is not in the document]
Comment (reszli) [reply src/_shared/ModalDialog.jsx]: not sure how to write that....
```
onClear: () => { 
    onAccept(null);
  }
```

File: src/_shared/ModalDialog.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmtrKovalenko) [reply src/_shared/ModalDialog.jsx]: Change tabIndex of your Cancel button to -2

File: src/_shared/ModalDialog.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmtrKovalenko) [reply src/_shared/ModalDialog.jsx]: You can pass default implementation from the root component, smth like
`onClear={onClear || this.handleClear}`
`this.handleClear = () => this.props.onChange(null)` <- put it to the PickerBase

**Comment by mfolnovic** [comment]
@reszli will you be able to update PR soon? 😊 

thanks!

**Comment by dmtrKovalenko** [comment]
You cannot? O.o I have reviewed it, and you should be able to update it

**Comment by mfolnovic** [comment]
@dmtrKovalenko wrong person, I was just pinging author of this PR. :)

**Comment by santosguilherme** [comment]
@mfolnovic nobody will see it?

**Comment by dmtrKovalenko** [comment]
@santosguilherme 
Seems its just forgotten 😢 
I left it to keep it open to remember me implement this feature by myself in the future

**Comment by santosguilherme** [comment]
@dmtrKovalenko 
Who would have to do the merge? Are you or @mfolnovic?

**Comment by dmtrKovalenko** [comment]
I am, but this pull request will not be merged, until the update, accordingly my review

**Comment by santosguilherme** [comment]
What do you think of the clear button being like the [Input Adornments](https://material-ui-next.com/demos/text-fields/#input-adornments)? I can try to do this.

**Comment by dmtrKovalenko** [comment]
I think if we should create some button, it should be inside the modal. 
P.S. Clear by input adornment can be created from the user side, with help of `InputProps`.

**Comment by santosguilherme** [comment]
Thanks!

Can I update this pull request for you or would it have to do a new PR?

**Comment by dmtrKovalenko** [comment]
Yep, feel free to create a new PR 😄 
P.S. You cannot update this PR, you should create a new one

**Comment by santosguilherme** [comment]
Okay, I'll try do this after my lunch!

**Pull request closed without merging**